docs/DeveloperGuide.md

```diff
@@ -155,6 +155,77 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Highlight error feature
+
+This feature would automatically highlight (select) the part of the command that contains the error in the command box,
+so that the user can edit it more quickly.
+
+For example, if the user type `edit 1 p/12 c/CS2103T`, since the phone number is invalid (but the course code `CS2103T` is valid), then `12` will be highlighted, which speeds up the process of data entry because the user no longer have to press `[←]` several times to navigate to `12`.
+
+#### Implementation
+
+First, many functions that originally takes the command string, will now take `CommandString`, which is a class
+containing a string.
+
+Compare the old and the new sequence diagram below for the execution of a command. This is the old sequence diagram.
+
+![OldArchitectureSequenceDiagram](images/OldArchitectureSequenceDiagram.png)
+
+And this is the new sequence diagram.
```

Comment:
Do we need to compare the current version with the old version? I think we can just only use the current diagram and delete(not mention) the old one.

Reply:
I think if it describes what is **changed** (i.e. how the feature is implemented **on top of the existing code**), then it's fine. After all, the proposed undo/redo feature also shows it as a diff.